node/core/pvf/src/prepare/worker.rs

```diff
@@ -294,5 +374,50 @@ fn prepare_artifact(code: &[u8]) -> Result<CompiledArtifact, PrepareError> {
 	.map_err(|panic_payload| {
 		PrepareError::Panic(crate::error::stringify_panic_payload(panic_payload))
 	})
-	.and_then(|inner_result| inner_result)
+	.and_then(|inner_result| inner_result)?;
+
+	let cpu_time_elapsed = cpu_time_start.elapsed();
```

Comment:
Why did you decide to move this code into `prepare_artifact`? To me it clearly belongs to some worker-related code, now it's unclear what `prepare_artifact` is responsible for (ditto for execute part).

Reply:
Yeah, I wasn't sure about this change. In the execute part we were already getting the elapsed duration, which is a necessary part of the response returned from that function. And I wanted to put the lock flag as close as possible to the end of the execute operation, so I stuck it in the middle of that function kind of awkwardly. Then I figured I could do the same in the prepare part, for parity with the execute code and also to reduce indentation (I believe it saved 4 indent levels).

I considered alternatively adding another function wrapper around `prepare_artifact` called `worker_prepare_job` or something, which would have all this thread coordination logic in it. But I feel like it'd also be awkward to have this big chain of function calls.

I thought the present solution wasn't too bad, because `prepare_artifact` was not doing much previously -- it was basically just two function calls, to `prevalidate` and `prepare`. Felt like the least annoying option.

Reply:
I would've still went with [this](https://github.com/paritytech/polkadot/blob/8e5139f1224eb1bd1c0caa1221fca6230beff583/node/core/pvf/src/prepare/worker.rs) implementation, (personally) the readability was better despite indentation levels

Reply:
Okay, I'll change it back. Thanks for the feedback.